File: Year 2/Semester 4/Programming and Projection Environments/Week 9/ProtobuffProjectCSharp/model/ShowDTO.cs
using ProtobuffProject.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProtobuffProject.domain
{
    [Serializable]
    public class ShowDTO<ID> : Entity<ID>
    {
        private int available;
        private int unavailable;

        public ShowDTO(ID id, string title, string date, string place, int available, int unavailable, int idArtist)
        :base(id)
        {
            this.Title = title;
            this.Date = date;
            this.Place = place;
            this.available = available;
            this.unavailable = unavailable;
            this.IdArtist = idArtist;
        }

        public string Title { get; set;}
        public string Date { get; set; }
        public string Place { get; set; }

        public int Available
        {
            get => available;
            set => available = value;
        }

        public int Unavailable
        {
            get => unavailable;
            set => unavailable = value;
        }

        public int IdArtist { get; set; }

        public override string ToString()
        {
            return base.ToString() + " " + Title + " " + Date + " " + Place + " " + available + " " + unavailable + " " + IdArtist;
        }
    }
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 9/ProtobuffProjectJava/Domain/src/main/java/gRPCProject/domain/ShowDTO.java
package gRPCProject.domain;

import java.io.Serializable;
import java.time.LocalDate;

public class ShowDTO extends Entity<Integer>implements Serializable {
    private final String title;
    private final LocalDate date;
    private final String place;
    private final Integer available;
    private final Integer unavailable;
    private final Integer idArtist;

    public ShowDTO(Integer id, String title, LocalDate date, String place,
                   Integer available, Integer unavailable, Integer idArtist) {
        super(id);
        this.title = title;
        this.date = date;
        this.place = place;
        this.available = available;
        this.unavailable = unavailable;
        this.idArtist = idArtist;
    }

    public String getTitle() {
        return title;
    }

    public LocalDate getDate() {
        return date;
    }

    public String getPlace() {
        return place;
    }

    public Integer getAvailable() {
        return available;
    }

    public Integer getUnavailable() {
        return unavailable;
    }

    public Integer getIdArtist() {
        return idArtist;
    }

    @Override
    public String toString() {
        return "ShowDTO{" +
                "title='" + title + '\'' +
                ", date=" + date +
                ", place='" + place + '\'' +
                ", available=" + available +
                ", unavailable=" + unavailable +
                ", idArtist=" + idArtist +
                '}';
    }
}